src/pages/guides/javascript/how-to-transfer-spl-tokens-on-solana.md

```diff
@@ -0,0 +1,203 @@
+
+// You will need to us fs and navigate the filesystem to
+// load the wallet you wish to use via relative pathing.
+const walletFile = const imageFile = fs.readFileSync(
+    path.join(__dirname, './keypair.json')
+  )
+```
+
+## Key Accounts
+
+When transferring SPL Tokens on Solana we need to work out both the senders and recieves SPL Token Account addresses for the token we are trying to send.
+
+Token Account addresses are unique between coins and wallets so we need to use a helper function to determine what each account address is for the both the sender and the reciever.
+
+```ts
+// The address of the Token you want to transfer.
+const splToken = publicKey('111111111111111111111111111111')
+
+// The address of the wallet you want to transfer the Token to.
+const recipeientWallet = publicKey('22222222222222222222222222222222')
+
+// Find the associated token account for the SPL Token on the senders wallet.
+const sourceTokenAccount = findAssociatedTokenPda(umi, {
+  mint: splToken,
+  owner: umi.identity.publicKey,
+})
+
+// Find the associated token account for the SPL Token on the receivers wallet.
+const destinationTokenAccount = findAssociatedTokenPda(umi, {
+  mint: splToken,
+  owner: recipeientWallet,
+})
+```
+
+## Sending the SPL Tokens
+
+```ts
+import {
+  findAssociatedTokenPda,
+  mplToolbox,
+  transferTokens,
+} from '@metaplex-foundation/mpl-toolbox'
+import {
+  generateSigner,
+  publicKey,
+  signerIdentity,
+  sol,
+} from '@metaplex-foundation/umi'
+import { createUmi } from '@metaplex-foundation/umi-bundle-defaults'
+import { base58 } from '@metaplex-foundation/umi/serializers'
+
+const transferSplTokens = async () => {
+  const umi = createUmi('https://api.devnet.solana.com').use(mplToolbox())
+
+  const signer = generateSigner(umi)
+
+  umi.use(signerIdentity(signer))
+
+  // Airdrop 1 SOL to the identity
+  // if you end up with a 429 too many requests error, you may have to use
+  // the filesystem wallet method or change rpcs.
+  await umi.rpc.airdrop(umi.identity.publicKey, sol(1))
+
+  //
+  // Key Accounts
+  //
+
+  // The address of the Token you want to transfer.
+  const splToken = publicKey('111111111111111111111111111111')
+
+  // The address of the wallet you want to transfer the Token to.
+  const recipeientWallet = publicKey('22222222222222222222222222222222')
+
+  // Find the associated token account for the SPL Token on the senders wallet.
+  const sourceTokenAccount = findAssociatedTokenPda(umi, {
+    mint: splToken,
+    owner: umi.identity.publicKey,
+  })
+
+  // Find the associated token account for the SPL Token on the receivers wallet.
+  const destinationTokenAccount = findAssociatedTokenPda(umi, {
+    mint: splToken,
+    owner: recipeientWallet,
+  })
+
+  //
+  // Transfer SPL Token
+  //
+
+  const res = await transferTokens(umi, {
+    source: sourceTokenAccount,
+    destination: destinationTokenAccount,
+    amount: 10000, // amount of tokens to transfer*
```

Comment:
should this "*" point to something? if not we should remove it?

Reply:
ah, should probably have a *lamports reference!